azurerm/internal/services/appplatform/resource_arm_spring_cloud_config_server.go

```diff
@@ -0,0 +1,390 @@
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"spring_cloud_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"uri": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: azure.ValidateConfigServerURI,
+			},
+			"host_key": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"host_key_algorithm": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"label": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"password": {
+				Type:      schema.TypeString,
+				Optional:  true,
+				Sensitive: true,
+			},
+			"private_key": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"repositories": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validate.NoEmptyStrings,
+						},
+						"uri": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: azure.ValidateConfigServerURI,
+						},
+						"host_key": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"host_key_algorithm": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"label": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"password": {
+							Type:      schema.TypeString,
+							Optional:  true,
+							Sensitive: true,
+						},
+						"pattern": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						"private_key": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"search_paths": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						"strict_host_key_checking": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"username": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"search_paths": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"strict_host_key_checking": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"username": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceArmSpringCloudConfigServerCreate(d *schema.ResourceData, meta interface{}) error {
+	springCloudId := d.Get("spring_cloud_id").(string)
+	gitProperty := expandArmSpringCloudConfigServerGitProperty(d)
+
+	if err := modifySpringCloudConfigServer(d, meta, springCloudId, gitProperty); err != nil {
```

Comment:
We need to check if the resource already exists like other creat functions

Reply:
could we also no use a shared function, seems like we could have a CreateUpdate function and then a delete more tailored to the task